Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -50,13 +61,55 @@ void UPSSpotComponent::OnUnregister()
 	Super::OnUnregister();
 }
 
+// Called when the end game state was changed to toggle progression widget visibility.
+void UPSSpotComponent::OnGameStateChanged_Implementation(ECurrentGameState CurrentGameState)
+{
+	APlayerCharacter* PlayerCharacter = UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter();
+	if (!ensureMsgf(PlayerCharacter, TEXT("ASSERT: [%i] %s:\n'PlayerCharacter' is not valid!"), __LINE__, *FString(__FUNCTION__)))
+	{
+		return;
+	}
+	const FPlayerTag& PlayerTag = PlayerCharacter->GetPlayerTag();
+
+	if (GetMeshChecked().GetPlayerTag() == PlayerTag)
```

Comment:
Please add silent early return, to avoid nesting the switch of the if-condition
```.cpp
if (GetMeshChecked().GetPlayerTag() != PlayerTag)
{
// Is not current spot
return;
}

switch 
...
```

Reply:
done https://github.com/moinm3uw/ProgressionSystem/pull/59/commits/d94a2f348b9e3313fddd5c21a70c8079f0a18fb3